### ISSUE TITLE
[crash] Focus crashes when accessing Settings

### ISSUE THREAD

**Issue by lobontiumira** [mozilla-mobile/focus-android#4801]
## Steps to reproduce
1. Install Focus through https://github.com/mozilla-mobile/focus-android/releases page or https://firefox-ci-tc.services.mozilla.com/tasks/index/project.mobile.focus.release/latest.
2. Open the app, tap on the main three-dot menu - Settings.

### Expected behavior
The app's settings page opens.

### Actual behavior
The app crashes continuously. Even when selecting to Close the app from the crash dialog.

### Device information
* Android device: Xiaomi Mi 8 Lite (Android 9), Samsung Galaxy Note 8 (Android 9)
* Focus version: 8.15.0, GV 88.0.20210412175251
* [logcat Focus.txt](https://github.com/mozilla-mobile/focus-android/files/6308483/logcat.Focus.txt)

https://user-images.githubusercontent.com/40694158/114661240-5ca76580-9cff-11eb-9249-f5d6faf0e0f8.mp4

**Comment by pocmo** [comment]
```
Caused by: java.lang.NoSuchMethodError: No direct method <init>(Ljava/lang/String;Lmozilla/components/concept/fetch/Request$Method;Lmozilla/components/concept/fetch/MutableHeaders;Lkotlin/Pair;Lkotlin/Pair;Lmozilla/components/concept/fetch/Request$Body;Lmozilla/components/concept/fetch/Request$Redirect;Lmozilla/components/concept/fetch/Request$CookiePolicy;ZILkotlin/jvm/internal/DefaultConstructorMarker;)V in class Lmozilla/components/concept/fetch/Request; or its super classes (declaration of 'mozilla.components.concept.fetch.Request' appears in base.apk)
        at org.mozilla.telemetry.net.TelemetryClient.uploadPing(TelemetryClient.kt:23)
        at org.mozilla.telemetry.schedule.jobscheduler.TelemetryJobService.uploadPingsInBackground(TelemetryJobService.java:24)
        at org.mozilla.telemetry.schedule.jobscheduler.TelemetryJobService$UploadPingsTask.doInBackground(TelemetryJobService.java:3)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
2021-04-14 08:42:43.509 18897-19049/? I/mozac/CrashReporter: Received crash: UncaughtExceptionCrash(timestamp=1618378963509, throwable=java.lang.RuntimeException: An error occurred while executing doInBackground(), breadcrumbs=[], uuid=99765964-7813-4501-90f0-a6288100fec6)
```

**Comment by jonalmeida** [comment]
I took some time to look into this and it's the same issue as https://github.com/mozilla-mobile/fenix/issues/17585 but with `service-telemetry` as the dependency that has a mismatched version of `concept-fetch`.

We're using a very old version of `service-telemetry` from v57.0.9 that was compiled against an older version of [`concept-fetch`](https://github.com/mozilla-mobile/android-components/blob/v57.0.9/components/service/telemetry/build.gradle#L26).

Now that our latest version of AC 74.0.10 includes an updated API (a new private flag), the signature has changed, so at runtime we're unable to match against the new signature.

There are two possible solutions that I can think of:
 1. We disable telemetry entirely until we release a Focus refresh.
 2. We try to get service-telemetry to compile against the newer version.

I took a stab at the second option with this patch below to exclude the old version and instead use the newer one:
```diff
diff --git a/app/build.gradle b/app/build.gradle
index 92ac5b82..e5a6a43f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -159,12 +159,16 @@ dependencies {
     implementation "org.mozilla.components:browser-search:$mozilla_components_version"
     implementation "org.mozilla.components:browser-session:$mozilla_components_version"
     implementation "org.mozilla.components:browser-state:$mozilla_components_version"
-    implementation "org.mozilla.components:concept-engine:$mozilla_components_version"
+    implementation "org.mozilla.components:concept-engine:$mozilla_components_version", {
+        force = true
+    }
     implementation "org.mozilla.components:concept-fetch:$mozilla_components_version"
     implementation "org.mozilla.components:feature-customtabs:$mozilla_components_version"
     implementation "org.mozilla.components:lib-crash:$mozilla_components_version"
     implementation "org.mozilla.components:lib-fetch-httpurlconnection:$mozilla_components_version"
-    implementation "org.mozilla.components:service-telemetry:$mozilla_components_version_telemetry"
+    implementation "org.mozilla.components:service-telemetry:$mozilla_components_version_telemetry", {
+        exclude group: 'org.mozilla.components', module: 'concept-fetch'
+    }
     implementation "org.mozilla.components:support-ktx:$mozilla_components_version"
     implementation "org.mozilla.components:support-utils:$mozilla_components_version"
     implementation "org.mozilla.components:ui-autocomplete:$mozilla_components_version"
```
Unfortunately, it doesn't seem to work - we still hit the API mismatch.

I think our best option in that case is to disable telemetry, which is just the core ping until we can migrate to glean when we start releasing from master.

**Comment by jonalmeida** [comment]
Also, to clarify, the reason I didn't run into this issue during my sanity check was because I was testing on a debug build which doesn't have telemetry pings enabled so we never hit this runtime error.

**Comment by jonalmeida** [comment]
After speaking with product, we don't want to disable the telemetry we do still get as it can be a valuable reference.

Sebastian had a great idea to vendor `service-telemetry` into Focus since it is the only consumer of the library and is currently unsupported by the telemetry team either way.

**Comment by jonalmeida** [comment]
This should be fixed with 8.5.1.

**Comment by lobontiumira** [comment]
Verified as fixed in the Focus 8.15.1 build with Samsung Galaxy Note 8 (Android 9).
